specification/databricks/resource-manager/Microsoft.Databricks/stable/2024-05-01/databricks.json

```diff
@@ -0,0 +1,2016 @@
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "PublicNetworkAccess"
+          }
+        },
+        "requiredNsgRules": {
+          "type": "string",
+          "description": "Gets or sets a value indicating whether data plane (clusters) to control plane communication happen over private endpoint. Supported values are 'AllRules' and 'NoAzureDatabricksRules'. 'NoAzureServiceRules' value is for internal use only.",
+          "enum": [
+            "AllRules",
+            "NoAzureDatabricksRules",
+            "NoAzureServiceRules"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "RequiredNsgRules"
+          }
+        },
+        "defaultCatalog": {
+          "description": "Properties for Default Catalog configuration during workspace creation.",
+          "$ref": "#/definitions/DefaultCatalogProperties"
+        },
+        "isUcEnabled": {
+          "readOnly": true,
+          "description": "Indicates whether unity catalog enabled for the workspace or not.",
+          "type": "boolean"
+        },
+        "accessConnector": {
+          "description": "Access Connector Resource that is going to be associated with Databricks Workspace",
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string",
+              "format": "arm-id",
+              "description": "The resource ID of Azure Databricks Access Connector Resource.",
+              "x-ms-arm-id-details": {
+                "allowedResources": [
+                  {
+                    "type": "Microsoft.Databricks/accessConnectors"
+                  }
+                ]
+              }
+            },
+            "identityType": {
+              "type": "string",
+              "description": "The identity type of the Access Connector Resource.",
+              "enum": [
+                "SystemAssigned",
+                "UserAssigned"
+              ],
+              "x-ms-enum": {
+                "modelAsString": true,
+                "name": "IdentityType"
+              }
+            },
+            "userAssignedIdentityId": {
+              "type": "string",
+              "format": "arm-id",
+              "description": "The resource ID of the User Assigned Identity associated with the Access Connector Resource. This is required for type 'UserAssigned' and not valid for type 'SystemAssigned'.",
+              "x-ms-arm-id-details": {
+                "allowedResources": [
+                  {
+                    "type": "Microsoft.ManagedIdentity/userAssignedIdentities"
+                  }
+                ]
+              }
+            }
+          },
+          "required": [
+            "id",
+            "identityType"
```

Comment:
Would you require both these properties in PATCH request scenarios?

Reply:
It is required only when you provide "accessConnector" if you look at properties object accessConnector is optional one.

Reply:
Right, but you could document in id, that 'if id is provided, identityType must also be specified' (as X?)

Reply:
I think you still didn't understood, if properties.accessConnector is provided then both properties.accessConnector.id, properties.accessConnector.identityType is mandatory. So we want to make it clear that in API spec as well, documentation is loose way to tell the things to customer. Making it part of API spec is better way to document contract, so we choose to follow the later